### PR TITLE
Unauthenticated users bag

### PR DESCRIPTION
**Pull request by Meir-p**
Prevent from unauthenticated users from always receiving error (Because there is no id)

### CODE REVIEW

**Comment by elitan** [comment]
@gdangelo Would you mind checking this PR?

**Comment by Meir-p** [comment]
@elitan It actually didn’t fix it but I will add a video Of the problem:
https://user-images.githubusercontent.com/100231666/171193567-ecf383d0-b475-4991-844f-bbb296a2443c.mov

File: docs/docs/platform/quickstarts/nextjs.mdx [hunk 1]
```diff
@@ -669,24 +678,30 @@ Finally, we can run our GraphQL query using the `useQuery` hook and the current
...
-  if (loading) {
-    return null
+  const { isAuthenticated } = useAuthenticationStatus()
+  var user = null;
+  
+  if(isAuthenticated){
+    // highlight-end
+    const id = useUserId()
+    
+    const { loading, error, data } = useQuery(GET_USER_QUERY, {
```
[the remark this answers is not in the document]
Comment (Meir-p) [reply docs/docs/platform/quickstarts/nextjs.mdx:689]: definitely right

**Comment by gdangelo** [comment]
> @gdangelo Would you mind checking this PR?

Sure! I'm looking at it.

File: docs/docs/platform/quickstarts/nextjs.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gdangelo) [reply docs/docs/platform/quickstarts/nextjs.mdx:689]: I agree. The best solution is to check if the `id` variable is not falsy before running the GraphQL query. 

As @plmercereau said, the Apollo client provides a `skip` option to don't execute the query if this option is `true`.

@Meir-p could you please update your PR with the following code instead?

```js
const id = useUserId();
const { loading, error, data } = useQuery(GET_USER_QUERY, {
  variables: { id },
  skip: !id,
});
```

Thanks

**Review by gdangelo — CHANGES_REQUESTED** [review]
No need to use the `useAuthenticationStatus` hook.

We can simply check the `id` variable returned by the `useUserId` hook. If it is falsy, then we don't run the GraphQL query using the `skip` option.

@Meir-p could you please revert your changes and add the `skip` option to the `useQuery` call?

```js
const id = useUserId();
const { loading, error, data } = useQuery(GET_USER_QUERY, {
  variables: { id },
  skip: !id,
});
```

**Comment by gdangelo** [comment]
Finally, I've created a new PR for this issue → #659

**Comment by plmercereau** [comment]
@Meir-p many thanks for raising up the issue and for helping us to adjust the quickstart!
Thanks Grégory too!
Closing the PR

**Pull request closed without merging**